ValveResourceFormat/IO/FileExtract.cs

```diff
@@ -48,9 +48,40 @@ public static Span<byte> Extract(Resource resource)
                     break;
 
                 case ResourceType.Mesh:
-                    // Wrap it around a KV3File object to get the header.
-                    data = Encoding.UTF8.GetBytes(((BinaryKV3)resource.DataBlock).GetKV3File().ToString());
+                case ResourceType.Model:
+                case ResourceType.World:
+                case ResourceType.WorldNode:
+                {
+                    if (package != null && targetPath != null)
+                    {
+                        data = Array.Empty<byte>();
+                        var exporter = new GltfModelExporter
+                        {
+                            FileLoader = new BasicVpkFileLoader(package)
+                        };
+                        switch(resource.ResourceType)
+                        {
+                            case ResourceType.Mesh:
+                                exporter.ExportToFile(resource.FileName, targetPath, new Mesh(resource));
+                                break;
+                            case ResourceType.Model:
+                                exporter.ExportToFile(resource.FileName, targetPath, (Model)resource.DataBlock);
+                                break;
+                            case ResourceType.WorldNode:
+                                exporter.ExportToFile(resource.FileName, targetPath, (WorldNode)resource.DataBlock);
+                                break;
+                            case ResourceType.World:
+                                exporter.ExportToFile(resource.FileName, targetPath, (World)resource.DataBlock);
```

Comment:
That being said, this doesn't actually work, does it? I don't see it being saved to a file.